src/main/java/com/icthh/xm/ms/dashboard/repository/impl/permitted/ConfigPermittedRepository.java

```diff
@@ -0,0 +1,39 @@
+package com.icthh.xm.ms.dashboard.repository.impl.permitted;
+
+import com.icthh.xm.commons.permission.service.PermissionCheckService;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public abstract class ConfigPermittedRepository {
+
+    private final PermissionCheckService permissionService;
+
+    protected ConfigPermittedRepository(
+        PermissionCheckService permissionService) {
+        this.permissionService = permissionService;
+    }
+
+    public Boolean matchSpelExpression(String privilegeKey, Object obj) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String condition = permissionService.createCondition(authentication, privilegeKey, new PermissionTranslator());
+
+        if (StringUtils.isBlank(condition)) {
+            return Boolean.TRUE;
+        }
+
+        ExpressionParser expressionParser = new SpelExpressionParser();
+        Expression expression = expressionParser.parseExpression(condition);
```

Comment:
do we really need to parse SpEL for every invocation? Can we try to find a way for caching them? Or maybe Expression itself is thread-safe and designed for reuse?

Reply:
added spel compilation